python/ray/tune/search/sample.py

```diff
@@ -532,8 +532,15 @@ def sample(
     ):
         if not isinstance(random_state, _BackwardsCompatibleNumpyRng):
             random_state = _BackwardsCompatibleNumpyRng(random_state)
-        values = self.sampler.sample(domain, spec, size, random_state=random_state)
+
+        quantized_domain = Domain()
```

Comment:
shouldn't we deepcopy the existing domain instead of creating a new one? seems like we are losing subclass type here.